### PR TITLE
Adding schema generation for PRIMITIVE-NON ARRAY message types

### PR DESCRIPTION
**Pull request by chathurawidanage**

### CODE REVIEW

**Comment by chathurawidanage** [comment]
For keyed messages, if key/value are primitive non arrays, user doesn't has to specify a schema(lengths).

Same applies for non-keyed, if value is a primitive non array.

File: twister2/api/src/java/edu/iu/dsc/tws/api/tset/schema/KeyedSchema.java [hunk 1]
```diff
@@ -14,20 +14,41 @@
...
+    this.totalSize = totalSize;
+  }
+
   @Override
   public MessageType getDataType() {
     return dType;
   }
 
+  @Override
+  public int getTotalSize() {
```
Comment (supunkamburugamuve) [inline twister2/api/src/java/edu/iu/dsc/tws/api/tset/schema/KeyedSchema.java]: Shall we make this data size? Otherwise, it can be confusing.

**Comment by chathurawidanage** [comment]
It seems we are not using Task Level OpsConfigs(edu.iu.dsc.tws.task.impl.ops.AbstractOpsConfig) to configure edges at TSet level. Any particular reason for not doing that? 

Since this is the case, this optimization will not work with TSet yet.  Please don't merge.

**Comment by supunkamburugamuve** [comment]
@chathurawidanage I think @nirandaperera is setting these manually to the edge

**Pull request merged**